src/components/ItemList.svelte

```diff
@@ -59,12 +59,35 @@
   }
 
   function fullTextSearch(query, searchItems) {
+    let filteredSearchItems = searchItems;
+    let searchQuery = query;
+
+    if (query.startsWith("tags:") || query.startsWith("origin:")) {
+      const labelType = query.split(":")[0]; // e.g. "tags" or "origin"
+      // separate label query from search query
+      const [label, searchValue] = query.split(":")[1].split(" ");
+
+      filteredSearchItems = searchItems.filter(
+        (item) => item[labelType] && item[labelType].includes(label)
+      );
+
+      // returns all items of that label if no extra search term is specified
+      if (!searchValue) return filteredSearchItems;
+      searchQuery = searchValue;
+    }
+
```

Comment:
This doesn't work for a case like: `save tags:Logins`, since it only looks at the start of the query. 

I think we probably should split this up into whitespace delimited tokens, process them individually (much as you are now), then pull the (non-tags/origin) tokens back together into a new search string via a `.join(" ")`.

It's a bit of a hassle but I think it's worth it to have things work as one would expect in all cases.